src/js/handler/mxCellHighlight.js

```diff
@@ -197,8 +197,12 @@ mxCellHighlight.prototype.repaint = function()
 		}
 		else
 		{
-			this.shape.bounds = new mxRectangle(this.state.x - this.spacing, this.state.y - this.spacing,
-					this.state.width + 2 * this.spacing, this.state.height + 2 * this.spacing);
+			const x = this.state.text ? Math.min(this.state.x, this.state.text.boundingBox.x) : this.state.x;
```

Comment:
Is this needed with the new code?

Reply:
Yeah, I added this for consistency because the red highlighted bounds for an expensive node wouldn't match the width of the green highlighted bounds when the node was selected.